photon/src/main/java/org/pushingpixels/photon/api/transcoder/SvgBatchConverter.java

```diff
@@ -132,42 +124,53 @@ public static void main(String[] args) {
         System.out.println(
                 "******************************************************************************");
 
-        for (File file : inputFolder.listFiles((File dir, String name) -> name.endsWith(".svg"))) {
-            String svgClassName = outputClassNamePrefix
-                    + file.getName().substring(0, file.getName().length() - 4);
-            svgClassName = svgClassName.replace('-', '_');
-            svgClassName = svgClassName.replace(' ', '_');
-            String classFilename = outputFolderName + File.separator +
-                    svgClassName + outputFileNameExtension;
-
-            System.err.println("Processing " + file.getName());
-
-            try (PrintWriter pw = new PrintWriter(classFilename);
-                 InputStream templateStream = SvgBatchConverter.class
-                         .getResourceAsStream(templateFile)) {
-                final CountDownLatch latch = new CountDownLatch(1);
-
-                SvgTranscoder transcoder = new SvgTranscoder(file.toURI().toURL().toString(),
-                        svgClassName, languageRenderer);
-                transcoder.setPackageName(outputPackageName);
-                transcoder.setListener(new TranscoderListener() {
-                    public Writer getWriter() {
-                        return pw;
-                    }
-
-                    public void finished() {
-                        latch.countDown();
-                    }
-                });
-                if (templateStream == null) {
-                    System.err.println("Couldn't load " + templateFile);
+        try (Stream<Path> pathStream = Files.walk(inputFolder, maxDepth)) {
+            pathStream.filter(path -> path.toString().endsWith(".svg")).forEach(path -> {
+                final String filename = path.getFileName().toString();
+                final String svgClassName = (outputClassNamePrefix + filename.substring(0, filename.length() - ".svg".length()))
+                        .replace('-', '_')
+                        .replace(' ', '_');
+
+                // the image ${inputFolder}/foo/bar/baz.svg is written to ${inputFolder}/foo/bar/${outputClassNamePrefix}_baz.java
+                // the class ${outputClassNamePrefix}_baz is placed in package ${outputPackageName}.foo.bar
+                final Path relativeParent = inputFolder.relativize(path).getParent();
```

Comment:
I'm not the biggest fan of how complex the logic is with this change. Before it's a rather straightforward iteration over all SVG files in a single folder. Now there's a lot of complexity in here for no apparent gain other than using stream / map APIs.

I'd rather keep the logic simple and have multiple invocations of the main converter in the script than push this complexity into the converter.

If you have something non-trivial in scope in mind, it would be better to first discuss that in bug tracker beforehand.